code/modules/mob/living/living_defense.dm

```diff
@@ -257,27 +257,28 @@
 	smoke_contact(S)
 
 /mob/living/proc/smoke_contact(obj/effect/particle_effect/smoke/S)
-	var/protection = max(1 - get_permeability_protection() * S.bio_protection, 0)
+	var/bioProtection = max(1 - get_permeability_protection() * S.bio_protection, 0)
+	var/acidProtection = max(1 - get_soft_acid_protection(), 0)
+	var/acidHardProtection = get_hard_acid_protection()
 	if(CHECK_BITFIELD(S.smoke_traits, SMOKE_EXTINGUISH))
 		ExtinguishMob()
 	if(CHECK_BITFIELD(S.smoke_traits, SMOKE_BLISTERING))
-		adjustFireLoss(15 * protection)
+		adjustFireLoss(15 * bioProtection)
 		to_chat(src, span_danger("It feels as if you've been dumped into an open fire!"))
 	if(CHECK_BITFIELD(S.smoke_traits, SMOKE_XENO_ACID))
-		if(prob(25 * protection))
+		if(prob(25 * acidProtection))
 			to_chat(src, span_danger("Your skin feels like it is melting away!"))
-		adjustFireLoss(S.strength * rand(20, 23) * protection)
+		adjustFireLoss(max(S.strength * rand(20, 23) * acidProtection - acidHardProtection, 0))
```

Comment:
Hard armor is not flat DR since a long time. It is anti-AP.

Reply:
Acid gas has no AP variable so this seems like a reasonable work-around.

Reply:
?